docs/architecture/adr-027-deterministic-protobuf-serialization.md

```diff
@@ -8,7 +8,28 @@
 
 Proposed
 
-## Context
+## Abstract
+
+Fully deterministic structure serializaiton, which works across many languages / clients,
+is needed for structure signature use-case. We need to be sure that whenever we serialize
+and deserialize a data structure, no matter in which supported language, the raw bytes
+will stay the same. Protobufs are not deterministic.
```

Comment:
This is repeating the "Context paragraph" imo.

> Protobuf seralization is not unique...

Reply:
Right. I was updating this paragraph few time and didn't even notice when I started covering the `context` part. 

I've moved the things around.